uarray/tests/test_uarray.py

```diff
@@ -28,3 +28,26 @@ def be2_ua_func(f, a, kw):
     be_inner.__ua_function__ = be2_ua_func
     with ua.set_backend(be_outer), ua.set_backend(be_inner):
         assert mm2() is obj
+
+
+def _extractor():
+    return ()
+
+
+def _replacer(args, kwargs, dispatchables):
+    return (args, kwargs)
+
+
+def test_pickle_support():
+    import pickle
+
+    mm = ua.generate_multimethod(_extractor, _replacer, "ua_tests")
+    mm.attr = "hello"
```

Comment:
Do we need to allow setting attributes?

Reply:
Yes, mainly for things like `__name__`, `__doc__`.